tests/v2/validation/rbac/globalrolesv2/globalroles_v2_webhook_test.go

```diff
@@ -0,0 +1,316 @@
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type GlobalRolesV2WebhookTestSuite struct {
+	suite.Suite
+	client       *rancher.Client
+	session      *session.Session
+	cluster      *management.Cluster
+	clusterID    string
+	clusterCount int
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TearDownSuite() {
+	grw.session.Cleanup()
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) SetupSuite() {
+	grw.session = session.NewSession()
+
+	client, err := rancher.NewClient("", grw.session)
+	require.NoError(grw.T(), err)
+
+	grw.client = client
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(grw.T(), clusterName, "Cluster name to install should be set")
+	grw.clusterID, err = clusters.GetClusterIDByName(grw.client, clusterName)
+	require.NoError(grw.T(), err, "Error getting cluster ID")
+	grw.cluster, err = grw.client.Management.Cluster.ByID(grw.clusterID)
+	require.NoError(grw.T(), err)
+
+	clusterNames, err := clusters.ListDownstreamClusters(grw.client)
+	grw.clusterCount = len(clusterNames)
+	require.NoError(grw.T(), err)
+
+}
+
+func getCRTBFromGRBOwner(t *testing.T, client *rancher.Client, user *management.User, expectedCrtbCount int) (*v3.ClusterRoleTemplateBindingList, error, string) {
+	log.Info("Verify that the global role binding is created for the user.")
+	grbOwner, err := getGlobalRoleBindingForUser(client, user.ID)
+	require.NoError(t, err)
+	require.NotEmpty(t, grbOwner, "Global Role Binding not found for the user")
+
+	log.Info("Verify that the cluster role template bindings are created for the downstream clusters.")
+	crtbList, err := listClusterRoleTemplateBindingsForInheritedClusterRoles(client, grbOwner, expectedCrtbCount)
+
+	return crtbList, err, grbOwner
+
+}
+
+func (grw *GlobalRolesV2WebhookTestSuite) TestRemoveOwnerLabelRejectedByWebhook() {
+	subSession := grw.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with inheritedClusterRoles and verify removing the grb owner label is rejected.")
+	user, err := createGlobalRoleAndUser(grw.T(), grw.client, []string{roleOwner})
+	require.NoError(grw.T(), err)
+
+	crtbList, err, _ := getCRTBFromGRBOwner(grw.T(), grw.client, user, grw.clusterCount)
+	require.NoError(grw.T(), err)
+
+	for _, crtb := range crtbList.Items {
+		newLabels := crtb.Labels
+		if newLabels == nil {
+			newLabels = make(map[string]string)
+		}
+		delete(newLabels, crtbOwnerLabel)
+
+		crtbWithUpdatedLabels := &v3.ClusterRoleTemplateBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        crtb.Name,
+				Namespace:   crtb.ObjectMeta.Namespace,
+				Labels:      newLabels,
+				Annotations: crtb.Annotations,
+			},
+			ClusterName:       crtb.ClusterName,
+			UserName:          crtb.UserName,
+			RoleTemplateName:  crtb.RoleTemplateName,
+			UserPrincipalName: crtb.UserPrincipalName,
+		}
+
+		_, err = rbac.UpdateClusterRoleTemplateBindings(grw.client, crtb.Namespace, crtbWithUpdatedLabels)
```

Comment:
Can we not implement it like this instead of making a copy of the crtb object:
	delete(crtb.Labels, crtbOwnerLabel)
	_, err = rbac.UpdateClusterRoleTemplateBindings(grw.client, crtb.Namespace, crtb)

Reply:
In delete(crtb.Labels, crtbOwnerLabel), we are deleting the crtbOwnerLabel but its not assigned to the existing crtb. Once after copying the crtb and assigning the newLabels(Line 94), we are making sure the labels are updated in the crtb. let me know if that makes sense.